src/test/fuzz/merkle.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2025 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <consensus/merkle.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/util/str.h>
+#include <util/strencodings.h>
+
+#include <cassert>
+#include <cstdint>
+#include <vector>
+
+FUZZ_TARGET(merkle)
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+
+    // Generate a random number of transactions (0-1000 to keep it reasonable)
+    const size_t num_txs = fuzzed_data_provider.ConsumeIntegralInRange<size_t>(0, 1000);
+    std::vector<uint256> tx_hashes;
+    tx_hashes.reserve(num_txs);
+
+    // Create a CBlock with fuzzed transactions
+    CBlock block;
+    block.vtx.resize(num_txs);
+
+    for (size_t i = 0; i < num_txs; ++i) {
+        CMutableTransaction mtx;
+        // Add minimal valid content to make it a transaction
+        mtx.version = fuzzed_data_provider.ConsumeIntegral<uint32_t>();
+        mtx.nLockTime = fuzzed_data_provider.ConsumeIntegral<uint32_t>();
+
+        // Add at least one input with a properly constructed Txid
+        CTxIn txin;
+        std::vector<unsigned char> prevout_hash_bytes = fuzzed_data_provider.ConsumeBytes<unsigned char>(32);
+        if (prevout_hash_bytes.size() < 32) {
+            prevout_hash_bytes.resize(32, 0); // Pad with zeros if needed
+        }
+        uint256 prevout_hash;
+        memcpy(prevout_hash.begin(), prevout_hash_bytes.data(), 32);
+        txin.prevout = COutPoint(Txid::FromUint256(prevout_hash), fuzzed_data_provider.ConsumeIntegral<uint32_t>());
+        std::vector<unsigned char> script_sig_bytes = fuzzed_data_provider.ConsumeBytes<unsigned char>(40);
```

Comment:
I'm new to the fuzzing test framework, but does this essentially build a fixed size 40 byte scriptSig? 

It may be worth making this variable in size to test different weaknesses in the merkle tree.

Particularly if we can occasionally generate a 64 byte transaction, whose format you can read about [here](https://delvingbitcoin.org/t/great-consensus-cleanup-revival/710/73)) you may be able to make some assertions on the `mutated` field below.

Reply:
Thanks for the review! I pushed [cdd3ad0](https://github.com/bitcoin/bitcoin/pull/32243/commits/cdd3ad098c169bfba96c51bf5e5d77602ff97276)

Now I am using the [MAX_SCRIPT_SIZE](https://github.com/bitcoin/bitcoin/blob/master/src/script/script.h#L39-L40) since that is what I see for the [IsUnspendable](https://github.com/bitcoin/bitcoin/blob/master/src/script/script.h#L566-L574) function for CScript